### PR TITLE
New team rollout ideas

### LINKED ISSUE #286: Guide on how to roll out sqlfluff on a new project/team
Referenced in #265 

First:
> It's a new structure and I don't want to trigger any negative stereotypes about engineers caring too much about unimportant details and slowing down the ML work, if that makes sense. If it were just me or just engineers, I'd feel less nervous.

Then
> Full transparency: We're not using sqlfluff at all in production at tails.com yet. I'm using it to clean up a few projects, and I'm almost ready to put it in the CI pipeline for one part of one project. I think there's probably a place for a guide about how to roll out sqlfluff on a project without pissing people off. 

Starting to use sqlfluff on a new project is probably daunting. There should be a short how-to on how to do it without pissing people off (...too much). This is a placeholder for that.

Anybody who has thoughts on this - feel free to drop comments into here (including you @barrywhart  😄 ).

### PR DESCRIPTION
**Pull request by alanmcruickshank**
Resolves #286.

I've been meaning to write this for a while. Would love people's input on the thrust of it. In particular @NiallRees, who I know has done some thinking about this.

### CODE REVIEW

File: docs/source/teamrollout.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+   *just your SQL*.
+
+With that in mind, it's worth reminding ourselves what we're trying
+to achieve with a tool like this. A set of potential success criteria
+might be:
+
+#. **Faster comprehension and collaboration** by the team on a shared
+   codebase. This includes more effective (and more enjoyable) code
+   review on top of code which is easy to review and build upon.
+#. **Easier and faster onboarding** on new team members. By adopting
```
Comment (NiallRees) [inline docs/source/teamrollout.rst:21]:
```suggestion
#. **Easier and faster onboarding** for new team members. By adopting
```

File: docs/source/teamrollout.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+   a style which is clean and *consistent with other organisations*
+   we make it easier for new people to join the team.
+#. **Improved adoption of shared SQL** from other sources. If the SQL
+   found in open source projects is easy to read and *looks familiar*
+   then you're more likely to use it. This means more reusable code
+   across the industry.
+#. **Productive duscussions around style**. By defining your
```
Comment (NiallRees) [inline docs/source/teamrollout.rst:28]:
```suggestion
#. **Productive discussions around style**. By defining your
```

**Review by NiallRees — APPROVED** [review]
This is great. Really well thought out.

**Comment by NiallRees** [comment]
I had an idea this weekend to create a tool to help teams determine their own .sqlfluff config, a web app which understands the configuration options both globally and for each rule. There could be drop downs for each configuration option upon which an example query would be 'fixed' to the new config. I'll have a further think

File: docs/source/teamrollout.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+   organisation's style guide in code, it means you can version
+   control it, discuss changes and ultimately give a concrete output
+   to discussions over style.
+
+   *You like leading commas? Make a PR to .sqlfluff and let's*
+   *dicuss with the team what the implications would be*.
```
Comment (pwildenhain) [inline docs/source/teamrollout.rst:34]:
```suggestion
   *discuss with the team what the implications would be*.
```

**Review by pwildenhain — APPROVED** [review]
👏🏻 Fabulous

Do we also want to mention our suite of related tools like the VSCode plugin or the online formatter?

File: docs/source/teamrollout.rst [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+#. **Soft CI/CD** (warnings but no strict fails).
+#. **Hard CI/CD** (violations mean deployments fail).
+
+In each of these phases you have three levers to play with:
+
+#. Areas of the project.
+#. Depth of rules enforced (this might also include whether
+   to ignore parsing errors or not too).
+#. Whether to just lint changes (:ref:`diff-quality`),
+   or whether to lint existing code.
```
Comment (barrywhart) [inline docs/source/teamrollout.rst:74]:
```suggestion
   or whether to lint all the existing code as well.
```

**Review by barrywhart — COMMENTED** [review]
One small suggested change.

Also, our organization uses SQLFluff as a pre-commit hook. Should we cover that as well?

**Pull request merged**